kyo-scheduler-pekko/jvm/src/main/scala/kyo/scheduler/KyoExecutorServiceConfigurator.scala

```diff
@@ -0,0 +1,52 @@
+package kyo.scheduler
+
+import com.typesafe.config.Config
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.ThreadFactory
+import org.apache.pekko.dispatch.DispatcherPrerequisites
+import org.apache.pekko.dispatch.ExecutorServiceConfigurator
+import org.apache.pekko.dispatch.ExecutorServiceFactory
+
+/** A Pekko ExecutorServiceConfigurator that integrates Kyo's adaptive scheduling capabilities with Pekko's dispatcher system. The
+  * configurator enables Kyo's scheduler to handle all actor executions within your Pekko application, allowing it to make optimal thread
+  * utilization decisions by having full visibility of the workload.
+  *
+  * To use Kyo's scheduler in your Pekko application, configure it as the default dispatcher:
+  *
+  * {{{
+  * pekko.actor.default-dispatcher {
+  *   type = "Dispatcher"
+  *   executor = "kyo.scheduler.KyoExecutorServiceConfigurator"
+  * }
+  * }}}
+  *
+  * The configurator uses Kyo's scheduler singleton instance, allowing it to share resources and optimization decisions across the entire
+  * application. By handling all actor executions, it can efficiently adapt to varying workloads and system conditions, optimizing thread
+  * utilization across your entire application.
+  *
+  * For effective load management, use Kyo's admission control through Scheduler.get.reject() methods at the boundaries of your application
+  * where external work enters the system. See the Admission class documentation for details on admission control behavior and
+  * configuration.
+  *
+  * @param config
+  *   The dispatcher configuration from Pekko
+  * @param prerequisites
+  *   Core Pekko prerequisites for dispatcher creation
+  *
+  * @see
+  *   [[kyo.scheduler.Scheduler]] for details on the underlying scheduling capabilities and `reject` methods
+  * @see
+  *   [[kyo.scheduler.regulator.Admission]] for details on admission control behavior
+  * @see
+  *   [[org.apache.pekko.dispatch.ExecutorServiceConfigurator]] for the Pekko dispatcher interface
+  */
+class KyoExecutorServiceConfigurator(config: Config, prerequisites: DispatcherPrerequisites)
+    extends ExecutorServiceConfigurator(config, prerequisites) {
```

Comment:
can we configure the Kyo scheduler in detail? Seems it's just using the default one.

Reply:
Kyo's scheduler is designed to work as a JVM-global resource so it doesn't allow creating separate instances. The scheduler has [several configs](https://github.com/search?q=repo%3Agetkyo%2Fkyo+path%3A%2F%5Ekyo-scheduler%5C%2Fjvm-native%5C%2Fsrc%5C%2Fmain%5C%2Fscala%5C%2Fkyo%5C%2Fscheduler%5C%2F%2F+Config&type=code) that can be set via system properties but I'd recommend trying the defaults first.

Reply:
elegant